browser/js/cart/cart.controller.js

```diff
@@ -14,11 +16,10 @@ app.controller('CartController', function ($scope, cart, CartFactory) {
 	$scope.total = total();
 
 	$scope.update = function (item) {
-		if(item.quantity !== "") {
+		item.quantity = parseInt(item.quantity);
+		if(!isNaN(item.quantity)) {
 			CartFactory.updateItem(item.product._id, item.quantity)
 			.then(function () {
-				// item.quantity = item.newQuantity;
-				// item.newQuantity = "";
 				$scope.total = total();
 			});
 		}
```

Comment:
don't forget to catch errors -- I know all of this isn't totally finished though